qiskit/providers/backend_compat.py

```diff
@@ -132,6 +133,14 @@ def convert_to_target(
                             target[inst][(qubit,)].calibration = calibration_entry
                     elif qargs in target[inst]:
                         target[inst][qargs].calibration = calibration_entry
+            if inst not in target and inst not in name_mapping:
+                warnings.warn(
+                    f"Custom calibration for '{inst}' defined in 'defaults' is dropped "
+                    "during the conversion to a target. If you don't want that, "
+                    "supply a custom_name_mapping or "
+                    "add the custom calibration to a target after building the target.",
+                    UserWarning,
+                )
```

Comment:
```suggestion
                else:
                    warnings.warn(
                        f"Custom calibration for '{inst}' defined in 'defaults' is dropped "
                        "during the conversion to a target. If you don't want that, "
                        "supply a custom_name_mapping or "
                        "add the custom calibration to a target after building the target.",
                        UserWarning,
                    )
```
How does checking the `name_mapping` help to detect missing entry? Probably you can check [`calibration_entry.user_provided`](https://github.com/Qiskit/qiskit-terra/blob/58aceb59e9bc1e16f29b09402e1984b16d211da3/qiskit/pulse/calibration_entries.py#L100-L102) to selectively trigger the warning.

Reply:
Okey. Target entry cannot be generated without having `Instruction` mapping and this looks somewhat obvious. On the other hand, I think there is the case a user provided `custom_name_mapping` but calibration is missing because key name is not registered to the basis_gates. In this case instmap entry is also silently gone, but warning is not raise because it checks `name_mapping`.